terraform/jenkins-instances.tf

```diff
@@ -0,0 +1,28 @@
+module "devsecops_mgmt_jenkins_networking" {
+  source = "../ansible/roles/external/gsa.jenkins/terraform/modules/networking"
+
+  vpc_id = "${var.sg_vpc_id == "" ? data.terraform_remote_state.infrastructure_remote_state.mgmt_vpc_id : var.sg_vpc_id}"
```

Comment:
Hmm, having conditional logic in Terraform feels like an antipattern, especially in a repository like this one that isn't meant to be directly reusable. Thoughts @LinuxBozo?

Reply:
This was the part that @LinuxBozo and I were looking at last week. The intention here is to check to see if a specific VPC ID was given in the variables... and if not, then query the terraform remote state of the DevSecOps-Infrastructure backend to pick the mgmt VPC and fill in the value. As this repo is intended to specifically hook up the that infrastructure, this conditional logic made sense... I think ;)

Reply:
Yeah, if there is no reason for this to be re-usable outside of the framework, and only makes sense within the thing as a whole, no reason for conditional logic here, should always just grab from remote state.

Reply:
disagree, but only because there are two ways in which this repo can be applied 1. it creates the mgmt vpc in which case the output is stored and retrieved from remote state 2. the mgmt vpc already exists (BSP traditionally created or via cloud formation) and is therefore declared as a variable. So the logic seems appropriate @afeld @LinuxBozo No?

Reply:
@JJediny 💥  Yep, that's a spot on use case for the conditional.

Reply:
@JJediny Sort of correct. The intention for this deployment was to plug in to the DevSecOps-Infrastructure repo/deployment. If you used that to create the architecture, then the conditional would come into play. Otherwise, if you created a separate architecture and wanted to use this repo to deploy Jenkins, then you would specify the VPC ID and Subnet ID in the appropriate variables. This conditional does not actually create the VPC or subnet, it just chooses which one to use. It assumes that they exist... it just depends on whether you used the DevSecOps-Infrastructure deployment or not.